### PR TITLE
Add 12th variant

### PR DESCRIPTION
**Pull request by n-gb**

### CODE REVIEW

**Comment by ghostinshadow** [comment]
Correct solution) One thing to think about is expression on 12..14 lines. Do you know a way to split it out to be more readable?

**Review by oleksandr-kychun — COMMENTED** [review]
Correct solution) One thing to think about is expression on 12..14 lines. Do you know a way to split it out to be more readable?

File: homework.rb [hunk 1]
```diff
@@ -1,6 +1,15 @@
...
 
 def generate
-  # put your solution here (delete the line below)
-  rand(1000000).to_s + '-' + rand(1000000).to_s
+  # pattern: "ABCDEF-123456-98"
+  # chunk of additional code to add (hopefully) more functionality:
+  temp_str = ''
+  NUM_OF_DIGITS.times { temp_str += rand(0...10).to_s }
+  # the main code:
+  ('a'..'z').to_a.sample(NUM_OF_LETTERS).join.upcase + '-' +
```
Comment (NadiaGubernator) [inline homework.rb]: The most obvious one is using backslashes at the end of each part of expression, if it is what you've meant.

**Comment by n-gb** [comment]
The most obvious one is using backslashes at the end of each part of expression, if it's about style.
Or using new variable to add parts of pattern to it, each addition in a new line.
Or using same old temp_str, but add "if" to iterator to have dash in the right place.

File: homework.rb (hunk 1 above)
Comment (sveredyuk) [inline homework.rb]: It's a little bit hard to understand what is going here, try to DRY, wrap magic calls in variables and so on.

File: homework.rb [hunk 2]
```diff
@@ -1,6 +1,19 @@
-VARIANT = 0
+VARIANT = 12
+LETTERS_LENGTH = 6
+NUM_PART1_LENGTH = 6
+NUM_PART2_LENGTH = 2
+
+def random_num_str(str_length)
+  str = ''
```
Comment (sveredyuk) [inline homework.rb]: without str = ''

File: homework.rb [hunk 3]
```diff
@@ -1,6 +1,19 @@
...
+  str
+end
 
 def generate
-  # put your solution here (delete the line below)
-  rand(1000000).to_s + '-' + rand(1000000).to_s
+  # pattern: "ABCDEF-123456-98"
+  letters = ''
+  LETTERS_LENGTH.times { letters += ('A'..'Z').to_a.sample }
+  num1 = random_num_str(NUM_PART1_LENGTH)
```
Comment (sveredyuk) [inline homework.rb]: no var1 or var2... avoid using variables with digits

File: homework.rb [hunk 4]
```diff
@@ -1,6 +1,19 @@
-VARIANT = 0
+VARIANT = 12
+LETTERS_LENGTH = 6
+NUM_PART1_LENGTH = 6
+NUM_PART2_LENGTH = 2
+
+def random_num_str(str_length)
+  arr = Array.new
+  str_length.times { arr << rand(0...10) }
```
Comment (sveredyuk) [inline homework.rb]: x.times.map...

File: homework.rb [hunk 5]
```diff
@@ -1,6 +1,19 @@
...
+  arr.join
+end
 
 def generate
-  # put your solution here (delete the line below)
-  rand(1000000).to_s + '-' + rand(1000000).to_s
+  # pattern: "ABCDEF-123456-98"
+  letters = ''
+  LETTERS_LENGTH.times { letters += ('A'..'Z').to_a.sample }
```
Comment (sveredyuk) [inline homework.rb]: x.times.map... avoid variables-containers like `letters = ''`

**Review by sveredyuk — APPROVED** [review]

**Pull request closed without merging**